[Kuo Chun-Yuan] iP

kevinkuo0320 opened the pull request:
# Duke 

> A chatbot to make your life easier


Duke provides the following features :+1: :
- add tasks
- delete tasks
- update tasks

Use it by doing the steps below:
1. Download the [code.](www.github.com)
2. Configure it
3. Run it on your computer
4. ~~make it better~~

Current Status
- [x] Level 1 - 9
- [ ] Level 10 (Coming Soon)

Basic git command that might help you:
```
git status
git add
git commit
```

kevinkuo0320 committed:
level 1

kevinkuo0320 committed:
level 2

kevinkuo0320 committed:
level 3

kevinkuo0320 committed:
level 4

kevinkuo0320 committed:
level 5

kevinkuo0320 committed:
level 6

kevinkuo0320 committed:
organize files into packages

kevinkuo0320 committed:
fix bugs on AddCommand

kevinkuo0320 committed:
Level-7: create storage function

kevinkuo0320 committed:
Level-8: time formatter

kevinkuo0320 committed:
update command feature

kevinkuo0320 committed:
Merge branch 'branch-Level-7'

kevinkuo0320 committed:
update storage bugs

kevinkuo0320 committed:
refractor project into classes and packages

kevinkuo0320 committed:
Add unit test for Duke

kevinkuo0320 committed:
Add Javadoc to all the class

kevinkuo0320 committed:
Ensure code follow coding standard

kevinkuo0320 committed:
Level-9: add find function

seanlaiys commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+
+/**
+ * Main class for Duke
+ *
```
Would deleting this empty line better aid being in line with the coding standard?

seanlaiys commented on src/main/java/utility/UI.java:
```diff
@@ -0,0 +1,39 @@
+
+    Scanner scanner;
+
+    public UI(){
```
Would adding a white space after the bracket be better?

seanlaiys commented on src/main/java/utility/UI.java:
```diff
@@ -0,0 +1,39 @@
+        this.scanner = new Scanner(System.in);
+    }
+
+    public void startMessage(){
```
Same as above, would adding a white space after the bracket be better?

seanlaiys commented on src/main/java/utility/UI.java:
```diff
@@ -0,0 +1,39 @@
+        System.out.println(msg);
+    }
+
+    public String readLine(){
```
Would adding a white space after the bracket be better?

seanlaiys reviewed (commented):
Overall excellent job! Just some small things to address.

kevinkuo0320 commented:
> Overall excellent job! Just some small things to address.

thank you for the comment!

kevinkuo0320 committed:
Add gradle support

kevinkuo0320 committed:
Merge branch 'add-gradle-support'

kevinkuo0320 committed:
Level-10 JavaFx

kevinkuo0320 committed:
update save.text

kevinkuo0320 committed:
Merge branch 'branch-level-10'

EvaderFati commented on src/main/java/duke/utility/Parser.java:
```diff
@@ -0,0 +1,48 @@
+package duke.utility;
+
+import duke.command.*;
```
Just for the sake for stupid coding standard, listing all the command explicitly might be better.
[https://se-education.org/guides/conventions/java/basic.html#package-and-import-statements](url)

EvaderFati commented on src/main/java/duke/utility/Parser.java:
```diff
@@ -0,0 +1,48 @@
+        Command command = null;
+
+        switch(commandTokenizer.nextToken()){
+            case "list":
```
Should these case clauses not be indented?
<img width="1056" alt="Screenshot 2022-02-04 at 11 16 49 PM" src="https://user-images.githubusercontent.com/52398668/152553974-75d7fc93-504b-4682-bfc9-d59d0dfcf29e.png">
.

EvaderFati commented on src/main/java/duke/utility/UI.java:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Handle user input
+ */
+public class UI {
```
Is it better to write the class name in PascalCase?
[https://se-education.org/guides/conventions/java/basic.html#naming](url)

EvaderFati reviewed (commented):
Overall I love how you construct your code! It could be even better after adding JavaDoc for some of your public methods and focusing more on coding standards(although sometimes its annoying and time-consuming).

kevinkuo0320 committed:
Fix command: make command parse without case insensitive

command for user input is case sensitive.

A case insensitive for command is more user-friendly
because users should not bother the exact case of the keywords.

Let's,
* update the command class to use case-insensitive parsing
* fix the bugs where user input wrong indentation

kevinkuo0320 committed:
Add the following features
- help commands
- new Gui interface
- fix datetime format

kevinkuo0320 committed:
Merge pull request #2 from kevinkuo0320/extension-d-help

Create extension for help feature

kevinkuo0320 committed:
Improve code quality

kevinkuo0320 committed:
Merge pull request #3 from kevinkuo0320/branch-A-CodeQuality

Improve code quality

kevinkuo0320 committed:
Add assertion on command Matcher

kevinkuo0320 committed:
Improve GUI for Duke:
- Highlight errors text
- circle profile picture

kevinkuo0320 committed:
Merge pull request #4 from kevinkuo0320/A-BetterGui

Improve GUI for Duke:

kevinkuo0320 committed:
Add Ui.png for Duke

kevinkuo0320 committed:
Add User Guide

kevinkuo0320 committed:
Update README

kevinkuo0320 committed:
Add more comments & fix indentation issues
